hathor/transaction/base_transaction.py

```diff
@@ -752,7 +757,7 @@ def _update_feature_activation_bit_counts(self) -> None:
         from hathor.transaction import Block
         assert isinstance(self, Block)
         # This method lazily calculates and stores the value in metadata
-        self.get_feature_activation_bit_counts()
+        cast(Block, self).get_feature_activation_bit_counts()
```

Comment:
This `cast` is necessary for mypy to work even with the `assert isinstance` check above, but it'll be removed in the next PR.

Reply:
This is the issue: https://github.com/python/mypy/issues/16494